tools/src/launcher/Main.hs

```diff
@@ -128,12 +129,20 @@ data NodeData = NodeData
     , ndArgs    :: ![Text]
     , ndLogPath :: Maybe FilePath }
 
--- | Updater path, args, windows runner path, archive path
-data UpdaterData = UpdaterData
-    { udPath        :: !FilePath
-    , udArgs        :: ![Text]
-    , udWindowsPath :: Maybe FilePath
-    , udArchivePath :: Maybe FilePath
+data UpdaterData = WindowsUpdaterData WinUD | OSXUpdaterData OSXUD | UnsupportedOSUpdateData UnsupOSUD
+
+data UnsupOSUD = UnsupOSUD
+
+data WinUD = WinUD 
+    { wudExePath :: !FilePath -- Installer.exe
+    , wudArgs    :: ![Text]
+    , wudBatPath :: !FilePath -- Installer.bat
+    }
+
+data OSXUD = OSXUD
+    { oudOpenPath :: !FilePath -- /usr/bin/open
+    , oudArgs     :: ![Text] -- FW
+    , oudPkgPath  :: !FilePath -- Installer.pkg
     }
```

Comment:
Some comment about more abstract terms applies here.

`oudOpenPath` – executable to launch the installer (e. g. /usr/bin/open).
`oudArgs` – command line arguments passed to `oudOpenPath` executable.
`oudPkgPath` – path to installer package, passed to `oudOpenPath` (e. g. Installer.pkg).